Source/init.cpp

```diff
@@ -49,10 +37,35 @@ HANDLE hfopt1_mpq;
 HANDLE hfopt2_mpq;
 #endif
 
+namespace {
+
+HANDLE init_test_access(std::string *out_mpq_path, const char *mpq_name, const char *reg_loc, int dwPriority, int fs)
+{
+	HANDLE archive;
+	const std::string *paths[2] = {&GetBasePath(), &GetPrefPath()};
+	std::string mpq_abspath;
+	for (int i = 0; i < 2; i++) {
+		mpq_abspath = *paths[i] + mpq_name;
+#if !defined(__SWITCH__) && !defined(__AMIGA__)
+		if (SFileOpenArchive(mpq_abspath.c_str(), dwPriority, MPQ_FLAG_READ_ONLY, &archive)) {
+#else
+		if (SFileOpenArchive(mpq_abspath.c_str(), dwPriority, 0, &archive)) {
+#endif

```

Comment:
lets make a var for the flag so the ifdef affects a minimum of code.
```suggestion
		int flags = 0;
#if !defined(__SWITCH__) && !defined(__AMIGA__)
		flags |= MPQ_FLAG_READ_ONLY;
#endif
		if (SFileOpenArchive(mpq_abspath.c_str(), dwPriority, flags, &archive)) {
```

Reply:
Done